source/tests/system/nirfmxspecan_driver_api_tests.cpp

```diff
@@ -424,6 +445,201 @@ TEST_F(NiRFmxSpecAnDriverApiTests, BuildSpurString_ReturnsSpurString)
   EXPECT_SUCCESS(session, spur_string_response);
 }
 
+void close_session(const client::StubPtr& stub, const nidevice_grpc::Session& session)
+{
+  ni::tests::system::EXPECT_SUCCESS(client::close(stub, session, Boolean::BOOLEAN_FALSE));
+}
+
+void force_close_session(const client::StubPtr& stub, const nidevice_grpc::Session& session)
+{
+  ni::tests::system::EXPECT_SUCCESS(client::close(stub, session, Boolean::BOOLEAN_TRUE));
+}
+
+class NiRFmxSpecAnDriverApiResourceInitTests : public NiRFmxSpecAnDriverApiTests,
+                                               public ::testing::WithParamInterface<std::tuple<std::string, std::string>> {
+};
+
+// Implements a function object that can be used as the name generator
+// for parameterized test suites instantiated from NiRFmxSpecAnDriverApiResourceInitTests.
+class ResourceTuplePrinter {
+ public:
+  using ParamType = NiRFmxSpecAnDriverApiResourceInitTests::ParamType;
+  std::string operator()(const TestParamInfo<ParamType>& info)
+  {
+    return sanitize_param<0>(info.param) + "__" + sanitize_param<1>(info.param);
```

Comment:
May be worth adding a little special logic for the empty string params.
![image](https://user-images.githubusercontent.com/14166913/152544841-c2e6638e-7ce8-4d18-aae1-16c9c3e7d092.png)